### ISSUE TITLE
[Bug]: "Provider produced inconsistent result after apply" in aws_cognito_user_pool_client

### ISSUE THREAD

**Issue by MrBuddyCasino** [hashicorp/terraform-provider-aws#30268]
### Terraform Core Version

1.1.7

### AWS Provider Version

~> 4.55

### Affected Resource(s)

* module.cognito_oauth.aws_cognito_user_pool_client

### Expected Behavior

`apply` should have succeeded

### Actual Behavior

Terraform output the error message provided below. It looks like an "empty list vs null" issue.

### Relevant Error/Panic Output Snippet

```shell
│ Error: Provider produced inconsistent result after apply
│ 
│ When applying changes to
│ module.cognito_oauth.aws_cognito_user_pool_client.this["connectservices-oauth-dev"],
│ provider "provider[\"registry.terraform.io/hashicorp/aws\"]" produced an
│ unexpected new value: .logout_urls: was cty.SetValEmpty(cty.String), but
│ now null.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
```

### Terraform Configuration Files

```
resource "aws_cognito_user_pool_client" "this" {
  for_each = var.pool_clients

  name         = each.key
  user_pool_id = aws_cognito_user_pool.this.id

  generate_secret       = true
  access_token_validity = 60

  allowed_oauth_flows = [
    "code",
  ]
  allowed_oauth_flows_user_pool_client = true
  allowed_oauth_scopes = [
    "email",
    "openid",
  ]

  callback_urls = each.value.callback_urls
  logout_urls   = []

  enable_propagate_additional_user_context_data = false

  enable_token_revocation = true
  id_token_validity       = 60

  token_validity_units {
    access_token  = "minutes"
    id_token      = "minutes"
    refresh_token = "days"
  }

  explicit_auth_flows = [
    "ALLOW_REFRESH_TOKEN_AUTH",
    "ALLOW_USER_PASSWORD_AUTH",
    "ALLOW_USER_SRP_AUTH",
  ]

  prevent_user_existence_errors = "ENABLED"

  read_attributes = [
    "email",
    "email_verified",
  ]
  write_attributes = [
    "email",
  ]

  supported_identity_providers = [
    aws_cognito_identity_provider.this.provider_name,
  ]
}
```

### Steps to Reproduce

I'm not sure it is feasible to reproduce a full Cognito setup. The "null vs empty list" issue can probably easily be checked in the backing code.

### Would you like to implement a fix?

None

**Comment by alnemo** [comment]
Seeing same, provider update to 4.60 broke compatibility for all existing Cognito clients, giving unexpected new value error for attributes 
.allowed_oauth_flows
.callback_urls
.logout_urls
.supported_identity_providers

**Comment by bidego** [comment]
Worked for me forcing provider <= 4.59 as @alnemo mentioned

**Comment by half2me** [comment]
same issue here when I have:
```typescript
explicit_auth_flows = []
```

**Comment by sherifkayad** [comment]
I have the same issue when I have empty callback URLs e.g.:
```terraform
callback_urls = []
```

**Comment by YakDriver** [comment]
Fixed by #30459

**Comment by HarryBo112** [comment]
Hi, it still seems to give an error. I am using tf core 1.4.2 and aws provider 4.62.0.
I've also tried other combinations (upgraded to aws provider 4.63.0, downgraded tf core), but it still gives me:

`│ Error: Provider produced inconsistent result after apply
│ 
│ When applying changes to aws_cognito_user_pool_client.my_ui, provider
│ "provider[\"registry.terraform.io/hashicorp/aws\"]" produced an unexpected
│ new value: .token_validity_units: block count changed from 0 to 1.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵`

Anything I could try? Downgrading aws provider to 4.59.0 does not show this issue.
